src/shared_realm.cpp

```diff
@@ -175,7 +179,21 @@ void Realm::open_with_config(const Config& config,
             read_only_group = std::make_unique<Group>(config.path, config.encryption_key.data(), Group::mode_ReadOnly);
         }
         else {
-            history = realm::make_in_realm_history(config.path);
+            // FIXME: The SharedGroup constructor, when called below, will
+            // throw a C++ exception if server_synchronization_mode is
+            // inconsistent with the accessed Realm file. This exception
+            // probably has to be transmuted to an NSError.
```

Comment:
Just throw a C++ exception and let Realm Cocoa, Realm Java, etc. handles it. Maybe add a special exception for it.